Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -233,6 +233,47 @@ final class CacheAdvanceTests: XCTestCase {
         XCTAssertEqual(messages, Array(LorumIpsum.messages.dropFirst(2)) + [barelyLongerMessage])
     }
 
+    func test_append_inAnOverwritingCacheEvictsMinimumPossibleNumberOfMessages() throws {
+        let message: TestableMessage = "test-message"
+        let maxMessagesBeforeOverwriting = [message, message, message]
+        let cache = try createCache(
+            sizedToFit: maxMessagesBeforeOverwriting,
+            overwritesOldMessages: true)
+        for message in maxMessagesBeforeOverwriting {
+            try cache.append(message: message)
+        }
+
+        // All of our messages have been stored and our cache is full.
+
+        // The byte layout in the cache should now be as follows:
+        // [header][length|test-message][length|test-message][length|test-message]
+        //         ^ reading handle                                              ^ writing handle
+
+        // When we read messages, we read from the current position of the reading handle – which is at the start of the oldest persisted message –
+        // up until the current position of the writing handle – which is at the end of the newest persisted message. This algorithm implies that if
+        // the reading handle and the writing handle are at the same position in the file, then the file is empty. Therefore, when writing a message
+        // and overwriting, we must ensure that we do not accidently write a message such that the reading handle and the writing handle end up in
+        // the same position.
```

Comment:
So clear, and the visual is great!

Reply:
I had to write out the visual in order to understand how to write the test 😂. I'm glad it was helpful for you too!